tests/widgets_block.rs

```diff
@@ -211,3 +211,471 @@ fn widgets_block_renders_on_small_areas() {
         Buffer::with_lines(vec!["┌Test─"]),
     );
 }
+
+#[test]
+fn widgets_block_renders_title_top_left_all_borders() {
```

Comment:
Thanks for including tests. However, a lot of them share the same structure. Could we use a more data-driven approach here ? For example, in [widgets_paragraph_can_wrap_its_content](https://github.com/fdehau/tui-rs/blob/1028d39db0946736b3c11a7c641f425dbf269030/tests/widgets_paragraph.rs#L16), the common structure is extracted to a closure which is then called with all test inputs. Additionally, the `TestBackend` could maybe a bit smaller to have more concise buffer assertions.

Reply:
Ohhhh, that is so neat! Thank you for showing the closure example, I will try to do something similar. 
I will make `TestBackend` smaller.